fs/span-manager/span_manager.go

```diff
@@ -467,32 +444,41 @@ func (m *SpanManager) uncompressSpan(s *span, compressedBuf []byte) ([]byte, err
 	return bytes, nil
 }
 
-func (m *SpanManager) fetchAndCacheSpan(spanID compression.SpanID) ([]byte, error) {
-	s := m.spans[spanID]
-	compressedSize := s.endCompOffset - s.startCompOffset
-	compressedBuf := make([]byte, compressedSize)
-	err := m.fetchSpan(compressedBuf, spanID)
-	if err != nil && err != io.EOF {
-		return nil, err
-	}
-
-	err = s.setState(fetched)
-	if err != nil {
-		return nil, err
+// addSpanToCache adds contents of the span to the cache.
+func (m *SpanManager) addSpanToCache(spanID compression.SpanID, contents []byte, opts ...cache.Option) {
```

Comment:
This existed before your change, be we just totally swallow all errors related to caching which will definitely break in the future. I think this method should return an error.

Reply:
yeah agree.

Reply:
updated